logger.js

```diff
@@ -0,0 +1,116 @@
+function log(message) {
+    console.log(message);
+}
+
+function log2(...msgArgs) {
+    let message = msgArgs.join(" | ");
+    log(message);
+}
+
+function log3(...msgArgs) {
+    let date = new Date();
+    let dateStr = `${date.toLocaleDateString()}, ${date.toLocaleTimeString()}`;
+    log2(dateStr, ...msgArgs);
+}
+
+log("Hello World !");
+log2("Hello", "World", "!");
+log3("Hello", "World", "!");
+
+
+//Functional pattern
+
+let baseLogger = {};
+(function (context) {
+    context.log = function (message) {
+        console.log(message);
+    };
+})(baseLogger);
+
+let tableLogger = {};
+tableLogger.log = function (...msgArgs) {
+    let message = msgArgs.join(" | ");
+    baseLogger.log(message);
+};
+
+let tableTimeLogger = {};
+tableTimeLogger.log = function (...msgArgs) {
+    let date = new Date();
+    let dateStr = `${date.toLocaleDateString()}, ${date.toLocaleTimeString()}`;
+    tableLogger.log(dateStr, ...msgArgs);
+};
+
+baseLogger.log("Hello World !");
```

Comment:
I want to see the following code:
`let logger = new BaseLogger();`
`logger.log("some text");`
`logger = new TableLogger();`
`logger.log("some text", "other text");`